### DISCUSSION TITLE
[Dev Tools] <Type error: Cannot read properties of undefined (reading "push")

### DISCUSSION THREAD

**Discussion by Szczepoo13** [zkSync-Community-Hub/zksync-developers#736] [Dev Tooling] [answered]
### Environment

Testnet

### Select the Dev Tool you are using

Hardhat Plugins

### Provide the version of the tool (if applicable)

The newest

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I am trying to implement a local instance of aave-v3-core, after cloning: https://github.com/aave/aave-v3-core.  
I first I run: **npm i -D @matterlabs/hardhat-zksync** 
Then, I import it to the hardhat.config.ts file. Then I add networks and enable zksync: True option

What is going on later is, I am getting the following error while I am trying to compile contracts with (**npx hardhat compile --network zkSyncSepoliaTestnet**): 
![screenDep 1](https://github.com/user-attachments/assets/e6cd602a-dfa7-44ae-a9c5-214f3814d5f6)

So, what could be the reason? I am using a node version v18.20.4, I cannot go any higher since this instance of aave is not allowing go any further than version 18 (version 14/16/18 is allowed).
Thank you for any insights.
Regards,

### Additional Details

Node version:  v18.20.4

**Comment by alhenaai** [comment]
The error you're encountering, "Type error: Cannot read properties of undefined (reading 'push')", typically indicates an issue with accessing an array or list-like structure that hasn't been properly initialized or is undefined at the time of access. This can occur in various scenarios, including misconfigurations or compatibility issues within your development environment.

Here are a few steps and considerations to help troubleshoot and potentially resolve this issue:

1. **Check `hardhat.config.ts` Configuration:**
   - Ensure that the `networks` configuration in your `hardhat.config.ts` file is correctly set up. The `zksync: true` flag should be properly placed within the network configuration object. Here's an example of how it should look:
     ```typescript
     networks: {
       zkSyncSepoliaTestnet: {
         url: "https://sepolia.era.zksync.dev",
         ethNetwork: "sepolia",
         zksync: true,
         verifyURL: "https://explorer.sepolia.era.zksync.dev/contract_verification",
       },
     }
     ```
   - Make sure that all necessary fields are correctly filled and that there are no syntax errors in the file.

2. **Node.js Compatibility:**
   - Since you are using Node.js v18.20.4, ensure that all npm packages and dependencies are compatible with this version. Sometimes, newer versions of packages might not support older or even relatively recent Node.js versions.

3. **Plugin and Dependency Versions:**
   - Verify that the version of `@matterlabs/hardhat-zksync` and other related dependencies are compatible with each other and with the version of Hardhat you are using. You might want to check the specific versions that are known to work well together.

4. **Reinstall Node Modules:**
   - Sometimes, simply reinstalling your node modules can resolve hidden issues. Delete your `node_modules` folder and `package-lock.json` file, then run `npm install` again to ensure a clean slate.

5. **Review the Error Stack Trace:**
   - Look closely at the stack trace provided by the error message for any clues on which part of your configuration or code might be causing the issue. This can often point you directly to the problematic part of your setup.

6. **Hardhat Compilation Settings:**
   - Ensure that your Hardhat setup is correctly utilizing the `zksolc` compiler if required by the ZKsync environment. This might involve specific settings in your `hardhat.config.ts` for the compiler.

If after these checks the issue persists, it might be beneficial to look for similar issues reported by other developers in community forums or even raise a question with specific details about your setup and the error in developer communities or directly in the repository issues if it seems related to a bug.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by uF4No** [comment, marked as answer]
Looks like the version of `hardhat` in that repo is quite old and is not compatible with our plugin. Try updating it to `2.22.x` . We'll add that to our docs.

**Reply by Szczepoo13** [reply]
@uF4No, could you confirm that using Hardhat versions below 2.22.x won't work?
